netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -269,8 +276,18 @@ public LogId getLogId() {
 
   @Override
   public Attributes getAttrs() {
+    if (attributesForTest != null) {
+      return attributesForTest;
+    }
     // TODO(zhangkun83): fill channel security attributes
-    return Attributes.EMPTY;
+    return Attributes.newBuilder().setAll(negotiator.getAttributes()).build();
```

Comment:
I'd like the attributes to be created just once, instead of rebuilding it every time. That way it's obvious it won't change.